### PR TITLE
run.py: Introduce metadata files for runs

### PR DESCRIPTION
**Pull request by k0machi**
This commit adds a new file emitted by the matrix runs, intended to
provide metadata about the run for reporting tools that are executed
later, such as Argus. Currently, the metadata file always contains the
driver name (derived from result xml name), type (in this case it's
always python), and either failure_reason key (which contains exception
stack trace that happened during .run() or path to the resulting junit
xml file relative to the metadata file.

This commit is part of ongoing task to improve matrix pipeline reporting
to Argus.

Task: scylladb/argus#289

### CODE REVIEW

File: scripts/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 six>=1.6
 pytest==6.2.5
-PyYAML==5.4
+PyYAML==6.0
```
Comment (fruch) [inline scripts/requirements.txt:3]: this doesn't do anything.

**Comment by fruch** [comment]
@k0machi we have a confirmation this is working as expected ?

**Comment by k0machi** [comment]
> @k0machi we have a confirmation this is working as expected ?

Seems like it, I will need to modify the pipeline now to verify it completely.

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**